node/node.go

```diff
@@ -426,22 +430,38 @@ func (node *Node) validateShardBoundMessage(
 		return nil, nil, errors.WithStack(errNotRightKeySize)
 	}
 
-	key, err := bls_cosi.BytesToBLSPublicKey(
-		senderPubKeyViaWire,
-	)
+	kkey := string(senderPubKeyViaWire)
+	key, err, _ := node.blsPubKeyCache.Do(kkey, func() (interface{}, error) {
+		k, e := bls_cosi.BytesToBLSPublicKey(senderPubKeyViaWire)
+
+		if e != nil {
+			return nil, e
+		}
+
+		if !node.Consensus.IsValidatorInCommittee(k) {
```

Comment:
why do you need to cache this part? if you only cache the deserialization, then you don't even need to timer, as blskey deserialization will never change.

Reply:
This part shouldn't be cached, because the IsValidatorInCommittee result may change across epoch and if it's cached for 12 hours it will be returning wrong results. One solution is to clear the cache whenever there is a new epoch transition.

Reply:
good point

Reply:
moved it out